js/service.js

```diff
@@ -8,7 +8,7 @@ let defaultParams = {
 };
 
 /**
- * Serialize a POJO as a URL query string fragment
+ * Serialize a POJO(Plain Old JavaScript Object. A set of key-value pairs.) as a URL query string fragment
```

Comment:
this is such a bad name... A JavaScript object is arbitrarily complex, whereas this function expects a flat object. I would strongly recommend we rename this to `toQueryArguments()` or something, and rewrite this a little so it doesn't blow up in our face when you *do* pass in something complicated. 

Right now, this code assumes that `input[somekey]` serializes to a string just fine, which is quite untrue. While clever, this code has some issues:
```
  return Object.keys(pojo).reduce((previousValue, currentValue, index) => {
    return `${previousValue}${index !== 0 ? `&` : ``}${currentValue}=${pojo[currentValue]}`;
  }, `?`);
```
The two main issues here are that

1. If a key has value `null` or `undefined`, it should not end up in the argument list, but this code adds it anyway (but `false` *should* end up in query args), and
2. there is no test for `${pojo[currentValue]}` to see if it makes sense, so this code will happy generate `[Object object]` without any indication that something just went horribly wrong.

So I'd recommend we rewrite this to something that does what we need with a bit more control and clarity:
```
function toQueryPair(key, data) {
  let val = data[key];
  if (val === undefined || val === null) {
    return false;
  }
  let type = typeof val;
  switch(type) {
    case "object": return false;
    case "function": return false;
    default: return `${key}=${val.toString()}`
  }
}

function toQueryString(data) {
  const pairs = Object.keys(data).map( key => toQueryPair(key, data)).filter(e => !!e);
  const queryArguments = pairs.join('&');
  return `?${queryArguments}`;
}
```
This code is certainly longer, but at every step we control whether or not data that should (or should not) be allowed is converted, explicitly, to string form and packed as URL argument key=value pair.